packages/zowe-explorer/src/uss/USSTree.ts

```diff
@@ -947,9 +949,15 @@ export class USSTree extends ZoweTreeProvider implements IZoweTree<IZoweUSSTreeN
      * @param this (resolves ESlint warning about unbound methods)
      * @param doc A doc URI that was closed
      */
-    public static onDidCloseTextDocument(this: void, doc: vscode.TextDocument): void {
+    public static onDidCloseTextDocument(this: void, doc: vscode.TextDocument): Promise<void> {
         if (doc.uri.fsPath.includes(globals.USS_DIR)) {
-            updateOpenFiles(TreeProviders.uss, doc.uri.fsPath, null);
+            return ZoweSaveQueue.all().then(() => {
+                if (!vscode.workspace.textDocuments.find(({ uri }) => uri.fsPath === doc.uri.fsPath)?.isDirty) {
+                    // Remove document from cache only if there are no pending unsaved changes
+                    updateOpenFiles(TreeProviders.uss, doc.uri.fsPath, null);
+                    LocalFileManagement.removeRecoveredFile(doc);
+                }
+            });
```

Comment:
I noticed this logic is almost the same as in `DatasetTree` (except for the variable access in `TreeProviders`).
Could this be condensed into a single function, so it only needs to be maintained in one place moving forward?